ext/intl/breakiterator/codepointiterator_internal.cpp

```diff
@@ -84,7 +84,7 @@ UBool CodePointBreakIterator::operator==(const BreakIterator& that) const
 	}
 
 	const CodePointBreakIterator& that2 =
-		static_cast<const CodePointBreakIterator&>(that);
+		dynamic_cast<const CodePointBreakIterator&>(that);
```

Comment:
I ll have a better look later, and I m all for using safer C++ operators ; but careful with `dynamic_cast` it can possibly throw `std::bad_cast`, in theory it should not here but dynamic_cast does it at _runtime time_ .

Reply:
I looked up into icu source code, should be fine, you re downcasting a reference here anyway.

Reply:
This change is pointless, the typeid is already checked at line 82, so static_cast is fine

Reply:
ah that s right, missed it.